src/d/actor/d_a_npc_md.cpp

```diff
@@ -1629,41 +1629,64 @@ BOOL daNpc_Md_c::hitNpcAction(void* r29) {
 }
 
 /* 00008E54-00008FFC       .text setNormalSpeedF__10daNpc_Md_cFfffff */
-void daNpc_Md_c::setNormalSpeedF(f32 f1, f32 f2, f32 i_scale, f32 i_maxStep, f32 i_minStep) {
-    f32 f31 = mMaxNormalSpeed * f1;
-    int r3 = wallHitCheck();
-    if (r3 >= 0) {
-        s16 angle = current.angle.y + 0x8000 - mAcchCir[r3].GetWallAngleY();
+void daNpc_Md_c::setNormalSpeedF(f32 i_speedScale, f32 i_acceleration, f32 i_scale, f32 i_maxStep, f32 i_minStep) {
+    // Calculate the desired speed based on a scaling factor and the NPC's maximum normal speed
+    f32 desiredSpeed = mMaxNormalSpeed * i_speedScale;
+
+    // Check if Medli is colliding with a wall
+    int wallIndex = wallHitCheck();
+    
+    if (wallIndex >= 0) {
+        // Calculate the angle between Medli and the wall
+        s16 angle = current.angle.y + 0x8000 - mAcchCir[wallIndex].GetWallAngleY();
+
+        // If the angle is not too steep, reduce speed based on angle and slowdown factor
         if (abs(angle) < 0x4000) {
-            f31 *= (1.0f - l_HIO.m008.m1C * cM_scos(angle));
+            desiredSpeed *= (1.0f - l_HIO.m008.wallSlowdownFactor * cM_scos(angle));
         }
     }
+
     f32 targetSpeed;
     f32 maxStep;
-    f32 f28;
-    if (f31 < m3108) {
-        f32 temp2 = m3108 - f31;
-        if (temp2 > i_maxStep) {
+
+    // If slowing down (desired speed is less than current speed)
+    if (desiredSpeed < mCurrentSpeed) {
```

Comment:
Some of these comments seem excessive. For example, does "If slowing down (desired speed is less than current speed)" give any more information than the line of code `if (desiredSpeed < mCurrentSpeed) {`?

It's better to comment either the purpose of the entire function above it, or of individual lines that do things that aren't immediately intuitive, rather than commenting every single line of code. Variable names should handle documenting most lines of code.